tensorboard/webapp/widgets/histogram/histogram_util.ts

```diff
@@ -148,9 +148,27 @@ function rebuildBins(bins: Bin[], range: Range, binCount: number): Bin[] {
 }
 
 /**
- * Computes how much of the input bin's 'y' counts should be allocated to a new
- * range. For 0 width input bins, the allocation may be split in half across 2
- * bins.
+ * Computes how much of the input bin's 'y' counts should be allocated to this output bin.
+ *
+ * Where both bins have non-zero width, this computed by multiplying the input y value by
+ * the ratio of the width-wise overlap in the bins to the total width of the output bin.
+ * (This can be thought of redistributing the overlapping "area" of the bar in the input
+ * histogram across the full width of the output bin.)
+ *
+ * When the input bin has zero width (the output bin cannot have zero width by construction),
+ * we instead have to consider several cases depending on the open/closed-ness of the
+ * underlying intervals. If the zero width input bin has y value 0, the contribution is
+ * always 0. Otherwise, if zero width input bin has y value greater than 0, it must represent
+ * the closed interval [x, x]. In this case, it contributes the full value of y if and only
+ * if the output bin's interval contains x. This interval is the closed-open interval
+ * [resultLeft, resultRight), except if resultHasRightNeighbor is false, in which case it's
+ * the closed interval [resultLeft, resultRight].
+ *
+ * For example, assuming the middle bin has width 0 and count 20:
```

Comment:
I'd probably omit this example, especially since the input bins shown aren't technically a valid set of bins and counts - since there is a zero-width bin that isn't the final bin, so it must be a closed-open interval, but it has a non-empty count, which shouldn't be possible.

If we do keep an example, I'd suggest phasing it in terms of the specific arguments to this function, like having a table in which there are columns for the function arguments and for the function return value, and each row represents a specific case, to help illustrate the logic in the paragraph of description.  But that might also be better just implemented as a set of test cases.

Reply:
Removed.